controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -55,26 +55,35 @@
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionException;
 import java.util.concurrent.CompletionStage;
+import java.util.concurrent.atomic.AtomicInteger;
 import java.util.concurrent.atomic.AtomicLong;
 import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Predicate;
 import java.util.stream.Collectors;
 import java.util.stream.IntStream;
 
-import static io.pravega.controller.store.stream.records.HistoryTimeSeries.HISTORY_CHUNK_SIZE;
-import static io.pravega.controller.store.stream.records.SealedSegmentsMapShard.SHARD_SIZE;
 import static io.pravega.shared.segment.StreamSegmentNameUtils.computeSegmentId;
 import static io.pravega.shared.segment.StreamSegmentNameUtils.getSegmentNumber;
 import static java.util.stream.Collectors.toMap;
 
 @Slf4j
 public abstract class PersistentStreamBase implements Stream {
-
     private final String scope;
     private final String name;
+    private final AtomicInteger historyChunkSize;
+    private final AtomicInteger shardSize;
 
     PersistentStreamBase(final String scope, final String name) {
         this.scope = scope;
         this.name = name;
+        historyChunkSize = new AtomicInteger(HistoryTimeSeries.HISTORY_CHUNK_SIZE);
+        shardSize = new AtomicInteger(SealedSegmentsMapShard.SHARD_SIZE);
+    }
+
+    @VisibleForTesting
+    void setHistoryChunkAndSealedSizeMapShardSizes(int historyChunkSize, int shardSize) {
```

Comment:
Even though this is flagged as visible for testing, it is opening up for changes to the internal state of the object, no?  Isn't there a better way of achieving the same goal, like a constructor that is visible for testing that takes these values as parameters?

Reply:
i can have the constructor, but we will need to add the constructor to the derived classes (ZkStream.java and InMemoryStream.java) as they are the classes that are actually instantiated. 
I was trying to avoid that change.. but i think its better to do it that way.. will make the change